src/helpers/radspec.js

```diff
@@ -1,39 +1,57 @@
-import ABI from 'web3-eth-abi'
-import { keccak256 } from 'web3-utils'
+import { ethers } from 'ethers'
+
 import MethodRegistry from './lib/methodRegistry'
 import { evaluateRaw } from '../lib/'
 import { knownFunctions } from '../data/'
+import { DEFAULT_API_4BYTES } from '../defaults'
 
 const makeUnknownFunctionNode = (methodId) => ({
   type: 'string',
   value: `Unknown function (${methodId})`
 })
 
-const getSig = (fn) =>
-  keccak256(fn).substr(0, 10)
+const parse = (signature) => {
+  const fragment = ethers.utils.FunctionFragment.from(signature)
+
+  return {
+    name:
+      fragment.name.charAt(0).toUpperCase() +
+      fragment.name
+        .slice(1)
+        .split(/(?=[A-Z])/)
+        .join(' '),
+    args: fragment.inputs.map((input) => {
+      return { type: input.type }
+    })
+  }
+}
+
+// Hash signature with Ethereum Identity and silce bytes
+const getSigHah = (sig) => ethers.utils.hexDataSlice(ethers.utils.id(sig), 0, 4)
 
 // Convert from the knownFunctions data format into the needed format
 // Input: { "signature(type1,type2)": "Its radspec string", ... }
-// Output: { "0xabcdef12": { "sig": "signature(type1,type2)", "source": "Its radspec string" }, ...}
-const processFunctions = (functions) => (
-  Object.keys(functions).reduce(
-    (acc, key) => (
-      {
-        [getSig(key)]: { source: functions[key], sig: key },
-        ...acc
-      }
-    ), {})
-)
-export default (eth, evaluator) =>
+// Output: { "0xabcdef12": { "fragment": FunctionFragment, "source": "Its radspec string" }, ...}
+const processFunctions = (functions) =>
+  Object.keys(functions).reduce((acc, key) => {
+    const fragment = ethers.utils.FunctionFragment.from(key)
+    return {
+      [getSigHah(fragment.format())]: { source: functions[key], fragment },
+      ...acc
+    }
+  }, {})
+
+export default (provider, evaluator) =>
   /**
    * Interpret calldata using radspec recursively. If the function signature is not in the package's known
    * functions, it fallbacks to looking for the function name using github.com/parity-contracts/signature-registry
    *
    * @param {address} addr The target address of the call
    * @param {bytes} data The calldata of the call
+   * @param {string} [registryAddress] The registry address to lookup descriptions
    * @return {Promise<radspec/evaluator/TypedValue>}
    */
-  async (addr, data) => {
+  async (addr, data, registryAddress) => {
```

Comment:
What you think of including an optional registry address to set up the Method Registry?